python-checks/src/main/java/org/sonar/python/checks/ConfusingTypeCheckingCheck.java

```diff
@@ -218,6 +204,20 @@ String message(String result) {
     }
   }
 
+  private static class NonCallableCalledCheck extends NonCallableCalled {
+
+    @Override
+    protected boolean isTypeSourceMatches(SubscriptionContext ctx, PythonType calleeType) {
+      return ctx.typeChecker().typeCheckBuilder().isTypeHintTypeSource().check(calleeType) == TriBool.TRUE;
+    }
+
+    @Override
+    protected boolean isNonCallableCall(SubscriptionContext ctx, PythonType calleeType) {
```

Comment:
`isNonCallableType` would make more sense here, as you're checking a type and not a call.